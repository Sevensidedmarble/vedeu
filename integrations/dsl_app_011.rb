#!/usr/bin/env ruby

# frozen_string_literal: true

require 'bundler/setup'
require 'vedeu'

TESTCASE = 'dsl_app_011'

class DSLApp

  Vedeu.bind(:_initialize_) { Vedeu.trigger(:_refresh_) }

  Vedeu.configure do
    debug!
    height 10
    log '/tmp/vedeu_views_dsl.log'
    renderers [
                Vedeu::Renderers::Terminal.new(
                  filename: "/tmp/#{TESTCASE}.out",
                  write_file: true),
                # Vedeu::Renderers::JSON.new(filename: "/tmp/#{TESTCASE}.json"),
                # Vedeu::Renderers::HTML.new(filename: "/tmp/#{TESTCASE}.html"),
                # Vedeu::Renderers::Text.new(filename: "/tmp/#{TESTCASE}.txt"),
              ]
    run_once!
    standalone!
  end

  load File.dirname(__FILE__) + '/support/test_interface.rb'

  Vedeu.render do
    view(:test_interface) do
      background '#002200'
      foreground '#aaff00'
      line 'view->line (align: :left)',   { align: :left }
      line 'view->line (align: :centre)', { align: :centre }
      line 'view->line (align: :right)',  { align: :right }
    end
  end

  def self.start(argv = ARGV)
    Vedeu::Launcher.execute!(argv)
  end

end # DSLApp

Vedeu.timer('Test') do
  DSLApp.start
end

load File.dirname(__FILE__) + '/test_runner.rb'
TestRunner.result(TESTCASE, __FILE__)
