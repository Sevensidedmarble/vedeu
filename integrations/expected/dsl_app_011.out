[0m[0m[2;2H[39m[49m(0l(B[2;3H(0q(B[2;4H [2;5HT[2;6He[2;7Hs[2;8Ht[2;9H [2;10H(0q(B[2;11H(0q(B[2;12H(0q(B[2;13H(0q(B[2;14H(0q(B[2;15H(0q(B[2;16H(0q(B[2;17H(0q(B[2;18H(0q(B[2;19H(0q(B[2;20H(0q(B[2;21H(0q(B[2;22H(0q(B[2;23H(0q(B[2;24H(0q(B[2;25H(0q(B[2;26H(0q(B[2;27H(0q(B[2;28H(0q(B[2;29H(0q(B[2;30H(0q(B[2;31H(0q(B[2;32H(0k(B[0m[3;2H[39m[49m(0x(B[3;3H[38;5;154m[48;5;16mv[3;4Hi[3;5He[3;6Hw[3;7H-[3;8H>[3;9Hl[3;10Hi[3;11Hn[3;12He[3;13H [3;14H([3;15Ha[3;16Hl[3;17Hi[3;18Hg[3;19Hn[3;20H:[3;21H [3;22H:[3;23Hl[3;24He[3;25Hf[3;26Ht[3;27H)[3;28H [3;29H [3;30H [3;31H [3;32H[39m[49m(0x(B[0m[4;2H[39m[49m(0x(B[4;3H[38;5;154m[48;5;16m [4;4Hv[4;5Hi[4;6He[4;7Hw[4;8H-[4;9H>[4;10Hl[4;11Hi[4;12Hn[4;13He[4;14H [4;15H([4;16Ha[4;17Hl[4;18Hi[4;19Hg[4;20Hn[4;21H:[4;22H [4;23H:[4;24Hc[4;25He[4;26Hn[4;27Ht[4;28Hr[4;29He[4;30H)[4;31H [4;32H[39m[49m(0x(B[0m[5;2H[39m[49m(0x(B[5;3H[38;5;154m[48;5;16m [5;4H [5;5H [5;6Hv[5;7Hi[5;8He[5;9Hw[5;10H-[5;11H>[5;12Hl[5;13Hi[5;14Hn[5;15He[5;16H [5;17H([5;18Ha[5;19Hl[5;20Hi[5;21Hg[5;22Hn[5;23H:[5;24H [5;25H:[5;26Hr[5;27Hi[5;28Hg[5;29Hh[5;30Ht[5;31H)[5;32H[39m[49m(0x(B[0m[6;2H[39m[49m(0m(B[6;3H(0q(B[6;4H(0q(B[6;5H(0q(B[6;6H(0q(B[6;7H(0q(B[6;8H(0q(B[6;9H(0q(B[6;10H(0q(B[6;11H(0q(B[6;12H(0q(B[6;13H(0q(B[6;14H(0q(B[6;15H(0q(B[6;16H(0q(B[6;17H(0q(B[6;18H(0q(B[6;19H(0q(B[6;20H(0q(B[6;21H(0q(B[6;22H(0q(B[6;23H(0q(B[6;24H(0q(B[6;25H(0q(B[6;26H(0q(B[6;27H(0q(B[6;28H(0q(B[6;29H(0q(B[6;30H(0q(B[6;31H(0q(B[6;32H(0j(B[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m